File: Prepend_and_Append.cpp
// we have a binary string s
// add 1 to one ending and 0 to other end of the string
// we need to find the length of the shortest possible string we could have started with
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;

int main(){
    ll testcases;
    cin >> testcases;

    for (ll testcase = 0; testcase < testcases; testcase++){
        ll n;
        cin >> n;

        string s;
        cin >> s;

        ll removable = 0;
        for (ll i = 0; i < n / 2; i++){
            if((s[i] == '1' && s[n - i - 1] == '0') || (s[i] == '0' && s[n - i - 1] == '1')){
                removable += 2;
            }
            else
                break;
        }

        cout << n - removable << endl;
    }
}

File: Prepend_and_Append.py
# we have a binary string s
# add 1 to one ending and 0 to other end of the string
# we need to find the length of the shortest possible string we could have started with
def main():
    testcases = int(input())

    for testcase in range(testcases):
        n = int(input())
        s = input()

        removable = 0
        for i in range(n // 2):
            if (s[i] == '1' and s[n - i - 1] == '0') or (s[i] == '0' and s[n - i - 1] == '1'):
                removable += 2
            else:
                break

        print(n - removable)

if __name__ == "__main__":
    main()